nmount: make `pmount` optional

Anomalocaridid opened the pull request:
Edit: Rather than entirely replacing pmount, this PR will now make pmount optional and use `udisksctl` if it is not installed.

Currently, `nmount` uses `pmount` for mounting and unmounting drives and `udisksctl` for powering them off. However, `udisksctl` can handle mounting and unmounting drives as well, so a dependency on `pmount` seems somewhat redundant, barring a reason to use `pmount` specifically.

In case it matters, this has the side effect of changing the default mount location from `/media/<device>` to udisks2's mount directory (which defaults to `/run/media/<username>/<volume label>`), although one can set a udev rule to change it back if desired: https://wiki.archlinux.org/title/Udisks#Mount_to_/media.

However, this side effect also makes `nmount` work better with other `udisks2`-based programs such as [`udiskie`](https://github.com/coldfix/udiskie), which uses `udisks2` to automatically mount removable drives. Currently, `nmount` does not play well with drives mounted by `udiskie`; attempting to unmount them causes an error.

N-R-K commented:
See this https://github.com/jarun/nnn/discussions/1667#discussioncomment-6137278

Anomalocaridid commented:
I see. I apologize for neglecting to check the discussions for similar topics.

In that case, how about I change my PR to instead make `pmount` optional and have `nmount` fall back to `udisksctl` if it is not installed?

jarun commented:
> In that case, how about I change my PR to instead make pmount optional and have nmount fall back to udisksctl if it is not installed?

I'm fine with that. It seems you are using 8 spaces instead of 4.
Also, does `udisksctl` work with LUKS encrypted volumes?

Anomalocaridid commented:
> I'm fine with that. It seems you are using 8 spaces instead of 4.

Sorry about that. I have fixed the formatting.

> Also, does `udisksctl` work with LUKS encrypted volumes?

Yes, however unlike `pmount`, it requires running separate commands for (un)mounting and (un)locking. I have addressed this by running those commands for all drives mounted using `udisksctl` and silencing the error output for the (un)locking commands. As far as I was able to determine, all the (un)locking commands do for non-LUKS volumes is exit with an error message.

I am unsure if this is the best approach for handling LUKS and non-LUKS volumes, so I added it in a separate commit so it can be easily removed or replaced if necessary.

jarun commented on plugins/nmount:
```diff
@@ -34,14 +34,30 @@ while [ -n "$dev" ]; do
+            else
+                udisksctl unmount -b "/dev/$dev" --no-user-interaction
+                exit_code="$?"
+                udisksctl lock -b "/dev/$dev" --no-user-interaction 2>/dev/null
```
We can't attempt lock/unlock on volumes without identifying if they are LUKS encrypted or not.
Please check the following link for methods: https://www.baeldung.com/linux/check-disk-encryption

Anomalocaridid replied:
I have re-implemented the locking and unlocking of LUKS volume with udiskctl so that it works properly. This time, I made sure to test my implementation with both un-encrypted and encrypted flash drives.

jarun commented:
Can someone else also test and confirm this change with udiskctl and pmount?

jarun commented:
Thank you!

Anomalocaridid commented:
Thank you as well

jarun commented:
I found it quite useful!
